src/test/ui/borrowck/borrowck-vec-pattern-nesting.stderr

```diff
@@ -36,12 +36,12 @@ LL |         &mut [_a,
    |
 help: consider removing the `&mut`
    |
-LL |         [_a,
-LL |
-LL |
-LL |
-LL |             ..
-LL |         ] => {
+LL ~         [_a,
+LL +
+LL +
+LL +
```

Comment:
Not sure why these empty lines are considered additions.

Reply:
That's what the follow up commits that I removed were addresssing: the highlighting mechanism is dump: it doesn't actually attempt to calculate a diff. What it _does_ is mark any whole line that is part of the replacement snippet as an addition, meaning that if the suggestion is in the form of `format!("some addition {}", snippet)`, everything that was part of `snippet` is _also_ marked as an addition. I have some thoughts about how to improve that, but for now I think we can address this best by gradually using more accurate spans.